### DISCUSSION TITLE
When inserting large amounts of data

### DISCUSSION THREAD

**Discussion by ingale726** [milvus-io/milvus#21571] [Q&A and General discussion]
When I inserted 3.04 million pieces of data, the search speed was around 0.1 second, when I inserted 6.2 million pieces of data, the search speed changed to around 20 seconds, my video card had 16 gigabytes of memory, and I saw that when I loaded 6.2 million pieces of data, the video memory was around 12 gigabytes. Is 6.2 million up to the limit of milvus loading and how can I optimize the parameters to maintain the retrieval speed of 0.1 second?

**Comment by xiaofan-luan** [comment]
If your vector dim is 128, 6.2m takes more than 3G.
if it is 768 dim, then more than 20G.

One thing you can do is to use IVF_SQ8 or IVF_PQ, which will save your. 4 times memory  but the recall will decrease.

**Reply by xiaofan-luan** [reply]
With Milvus 2.2.2, even if you only use CPU, the search speed should be faster than 100 ms if you have 3m data.

**Reply by ingale726** [reply]
Looking forward to 2.2.2, and when will the 2.x version support Gpus?

**Reply by xiaofan-luan** [reply]
> Looking forward to 2.2.2, and when will the 2.x version support Gpus?

Not yet. we have a version but the GPU version didn't outperform  HNSW too much

**Reply by ingale726** [reply]
Now version 2.x doesn't work in production because it doesn't support Gpus, can version 2.2.2 work in production? As you said, it can retrieve results in 0.1 seconds using only CPU

**Reply by xiaofan-luan** [reply]
> Now version 2.x doesn't work in production because it doesn't support Gpus, can version 2.2.2 work in production? As you said, it can retrieve results in 0.1 seconds using only CPU

If CPU is fast enough, is there a reason we must use GPU? because GPU is much more expensive

**Reply by ingale726** [reply]
It makes sense
